### ISSUE TITLE
IDE0005 is not reported if a code have same name extension methods.

### ISSUE THREAD

**Issue by naminodarie** [dotnet/roslyn#54283]
**Version Used**: 

Microsoft Visual Studio Community 2019
Version 16.10.1

**Steps to Reproduce**:

```csharp
using System.Linq;
using D;

public class Program
{
    static void Main(string[] args)
    {
        1.Count();
    }
}

namespace D
{
    static class E
    {
        public static int Count(this int n) => n;
    }
}
```

Do "Remove and Sort Usings" in this code.

**Expected Behavior**:

IDE0005 is reported on `using System.Linq;`.
`using System.Linq;` is removed.

**Actual Behavior**:

Nothing changes.

------

All classes that have an extension method with same name are marked as used. 

https://github.com/dotnet/roslyn/blob/12ca809cd57e19117e7ba74d06e7f4e21fc22bfb/src/Compilers/CSharp/Portable/Binder/WithUsingNamespacesAndTypesBinder.cs#L72,L128

**Comment by naminodarie** [comment]
```csharp
using System.Linq;

public class Program
{
    static void Main(string[] args)
    {
        1.Count();
    }
}

static class E
{
    public static int Count(this int n) => n;
}
```

If class `E` is in global namespace, "Remove and Sort Usings" removes `using System.Linq;`.

**Comment by sharwell** [comment]
Moving to Area-Compilers. IDE0005 is reported on cases where the compiler indicates a using directive is unused, and it's not doing so in this case.

**Comment by AlekseyTs** [comment]
I do not believe there was any recent change around this behavior. The design is: "// If we found any extension methods, then consider this using as used." In other words, if we need to look for an extension method in a using and found a candidate, the using is considered used. We do not postpone the decision until overload resolution decides if the candidate is good. The same is pretty much happening for other scenarios, the ```MarkImportDirective``` helper is called during lookup, before overload resolution and other checks. Moving the logic for extension methods to a later phase, though possible, perhaps not worth complexity and the risk.
